feed2maildir/converter.py

```diff
@@ -55,6 +57,28 @@ def get_data(self):
                 out += '  [{}]: {}\n'.format(l, self.links[l])
         return out
 
+class ExternalHTMLStripper:
+    def __init__(self, strip_program):
+        self.strip_program = strip_program
+        self.reset()
+
+    def feed(self, data):
+        self.raw_data.append(data)
+
+    def close(self):
+        p = Popen(self.strip_program, stdin=PIPE, stdout=PIPE, shell=True)
```

Comment:
Any particular reason for using `Popen` over `subprocess`?

Reply:
Also, I think we might want to handle common failure cases, like program not found and non-zero exit codes.

Reply:
`subprocess.Popen` is flexible enough to allow me to send it data and receive from it the output via the `communicate` method.

If the program does not exist, python will race an exception, however if it fails with a non-zero exit code the user will not notice it. I'm going to add a check for that, print the stderr and raise an exception. Good catch!

Reply:
I did an extra check and if a program does not exist, `Popen` will not fail. This is because the program executed is actually a shell (`shell=True`). Checking for the exit code later will catch that error case.